packages/@react-aria/menu/src/useMenuTrigger.ts

```diff
@@ -45,7 +45,7 @@ export interface MenuTriggerAria<T> {
  * @param state - State for the menu trigger.
  * @param ref - Ref to the HTML element trigger for the menu.
  */
-export function useMenuTrigger<T>(props: AriaMenuTriggerProps, state: MenuTriggerState, ref: RefObject<Element>): MenuTriggerAria<T> {
+export function useMenuTrigger<T>(props: AriaMenuTriggerProps, state: MenuTriggerState, ref?: RefObject<Element>): MenuTriggerAria<T> {
```

Comment:
At a glance moving to using useOverlayTrigger seems feasible, pretty similar to what we do for MobileCombobox. Like you said, we don't need the long press stuff so we'd only need to copy over the id/pressProps stuff from useMenuTrigger into useComboBox.

Reply:
hi do you want me to make contribute useOverlayTrigger change.

Reply:
Definitely feel free to give it a shot if you'd like! It would be good to know if something I'm overlooking would breaks with that change, just haven't had time to try it myself

Reply:
oh okay so there is these 2 props - 
-  **aria-labelledby** which would break with using only overlay trigger and doesnt make sense to have arial labeled by in [overlay trigger ](https://react-spectrum.adobe.com/react-aria/useComboBox.html#usecombobox)
- **onKeyDown** - this im assuming is invoked whenever you press any key and it shows the combobox with relevant list item
Both of these props dont fit in useOverlayTrigger.
### We can do either of these - 
  - make non null assertion for the useMenuTrigger invocation in useCombobox
  - Keep the ref optional. 
  > I don't think useMenuTrigger should have its ref as optional, it would interfere with longpress handling and arrow keys.

On this point by @snowystinger would that not be true for arrow keys for combobox?  Ignoring the longpress handling which doesnt make sense for combobox
  - create a sepeate trigger hook that takes in relevant combobox hook
  solves all problem but with code duplication as the new hook would essentially by useMenuTrigger without the required prop

Reply:
apologies about the delay, mind clarifying these two:

> aria-labelledby which would break with using only overlay trigger and doesnt make sense to have arial labeled by in [overlay trigger](https://react-spectrum.adobe.com/react-aria/useComboBox.html#usecombobox)

Could we set up a `useId` call in useComboBox and use that for the trigger id and the menu's `aria-labelledby` and override the ones that will be provided by useOverlayTrigger?

> onKeyDown - this im assuming is invoked whenever you press any key and it shows the combobox with relevant list item

similar to above, perhaps we could copy over the menutrigger's onKeyDown over to useComboBox as well?

Reply:
hi sorry, been busy with work. will try your suggestions soon thanks for guidance.